src/lambda/wizards/samInitWizard.ts

```diff
@@ -80,6 +100,171 @@ export class DefaultCreateNewSamAppWizardContext extends WizardContext implement
+            if (semver.lt(this.samCliVersion, MINIMUM_SAM_CLI_VERSION_SCHEMAS_SUPPORT_INCLUSIVE)) {
+                vscode.window.showErrorMessage(
+                    localize(
+                        'AWS.samcli.error.invalid_schema_support_version',
+                        'Installed SAM executable does not support templates that require Event Schema selection. Required minimum version {0}, but found {1}',
+                        MINIMUM_SAM_CLI_VERSION_SCHEMAS_SUPPORT_INCLUSIVE,
+                        this.samCliVersion
+                    )
+                )
+
+                return exitWizard
+            }
+        }
+
+        return val ? (val.label as SamTemplate) : undefined
+    }
+
+    public async promptUserForRegion(currRegion?: string): Promise<string | undefined> {
+        const schemasRegions = ['us-east-1', 'us-east-2', 'us-west-2', 'eu-west-1', 'ap-northeast-1']
+
+        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
+            options: {
+                ignoreFocusOut: true,
+                title: localize('AWS.samcli.initWizard.schemas.region.prompt', 'Select a Region'),
+                value: currRegion ? currRegion : ''
+            },
+            buttons: [this.helpButton, vscode.QuickInputButtons.Back],
+            items: schemasRegions.map(region => ({
+                label: region,
+                alwaysShow: region === currRegion,
+                description:
+                    region === currRegion ? localize('AWS.wizard.selectedPreviously', 'Selected Previously') : ''
+            }))
+        })
+
+        const choices = await picker.promptUser({
+            picker: quickPick,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                } else if (button === this.helpButton) {
+                    vscode.env.openExternal(vscode.Uri.parse(samInitDocUrl))
+                }
+            }
+        })
+        const val = picker.verifySinglePickerOutput(choices)
+
+        return val ? (val.label as string) : undefined
+    }
+
+    public async promptUserForRegistry(currRegion: string, currRegistry?: string): Promise<string | undefined> {
+        const client: SchemaClient = ext.toolkitClientBuilder.createSchemaClient(currRegion)
+        const registryNames = await SchemasDataProvider.getInstance().getRegistries(currRegion, client)
+
+        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
+            options: {
+                ignoreFocusOut: true,
+                title: localize('AWS.samcli.initWizard.schemas.registry.prompt', 'Select a Registry'),
+                value: currRegistry ? currRegistry : ''
+            },
+            buttons: [this.helpButton, vscode.QuickInputButtons.Back],
+            items: registryNames!.map(registry => ({
```

Comment:
Will registryNames  ever contain zero or one item?

If zero, the picker wouldn't be able to proceed.
If one, is it worth showing? (I think so, but wanted to bring it up)

Reply:
It should never contain zero item as *aws.events *registry should exist in all available regions.  I have had a quick discussion about whether or not we should skip the registry selection if there is only one element present. At that point, we didn’t fully finalize a decision and decided to go as is, I will follow up on this and let you know.